File: python-basic/formless.py
import os
import asyncio
from dotenv import load_dotenv
from openai import AsyncOpenAI

load_dotenv()

async def main():
    messages = [
        {
            "role": "user",
            "content": "Hello. What is your name?",
        }
    ]

    try:
        # Check for SHAPESINC_USER_API_KEY in .env
        if not os.getenv("SHAPESINC_USER_API_KEY"):
            raise ValueError("SHAPESINC_USER_API_KEY not found in .env")

        # Create the client with the user API key and the Formless API base URL
        aclient_formless = AsyncOpenAI(
            api_key=os.getenv("SHAPESINC_USER_API_KEY"),
            base_url="https://api.shapes.inc/formless/",
        )

        # Send the message to the Formless LLM. This will use the Formless model.
        resp = await aclient_formless.chat.completions.create(
            model="shapesinc/Formless-70B-v1a",
            messages=messages,
        )
        print(f"Raw response: {resp}")

        if resp.choices and len(resp.choices) > 0:
            final_response = resp.choices[0].message.content
            print(f"Reply: {final_response}")
        else:
            print(f"No choices in response: {resp}")

    except Exception as e:
        print(f"Error: {e}")


if __name__ == "__main__":
    asyncio.run(main())

File: nodejs-basic/formless.js
import { config } from "dotenv";
import OpenAI from "openai";

config();

async function main() {
  try {
    // Check for SHAPESINC_USER_API_KEY in .env
    if (!process.env.SHAPESINC_USER_API_KEY) {
      throw new Error("SHAPESINC_USER_API_KEY not found in .env");
    }

    // Create the client with the user API key and the Formless API base URL
    const openai = new OpenAI({
      apiKey: process.env.SHAPESINC_USER_API_KEY,
      baseURL: "https://api.shapes.inc/formless/"
    });

    // If the user provided a message on the command line, use that one
    const args = process.argv.slice(2);
    const messages = [
      { role: "user", content: args.length > 0 ? args.join(" ") : "Hello. What is your name?" }
    ];

    // Send the message to the Formless LLM. This will use the Formless model.
    const resp = await openai.chat.completions.create({
      model: "shapesinc/Formless-70B-v1a",
      messages
    });

    console.log("Raw response:", resp);

    if (resp.choices && resp.choices.length > 0) {
      console.log("Reply:", resp.choices[0].message.content);
    } else {
      console.log("No choices in response:", resp);
    }
  } catch (error) {
    console.error("Error:", error);
  }
}

main();